let User,CustomErrorHandler;_295‍.x([["default",()=>_295‍.o]]);_295‍.w("../models",[["User",["User"],function(v){User=v}]]);_295‍.w("../services/CustomErrorHandler",[["default",["CustomErrorHandler"],function(v){CustomErrorHandler=v}]]);


const admin = async (req, res, next) => {
    try {
        const user = await User.findOne({ _id: req.user._id });
        if (user.role === 'admin') {
            next();
        } else {
            return next(CustomErrorHandler.unAuthorized());
        }
    } catch (err) {
        return next(CustomErrorHandler.serverError(err.message));
    }
};

_295‍.d(admin);
